adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/optypes/JDBCDMLWriteOp.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.nosqlbench.adapter.jdbc.optypes;
+
+import io.nosqlbench.adapter.jdbc.JDBCSpace;
+import io.nosqlbench.adapter.jdbc.exceptions.JDBCAdapterUnexpectedException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class JDBCDMLWriteOp extends JDBCDMLOp {
+    private static final Logger LOGGER = LogManager.getLogger(JDBCDMLWriteOp.class);
+
+    private final int ddlStmtBatchNum;
+
+    protected final static ThreadLocal<Integer> threadBatchTrackingCntTL = ThreadLocal.withInitial(() -> 0);
+
+    public JDBCDMLWriteOp(JDBCSpace jdbcSpace,
+                          boolean isReadStmt,
+                          String pStmtSqlStr,
+                          List<Object> pStmtValList,
+                          int ddlStmtBatchNum) {
+        super(jdbcSpace, isReadStmt, pStmtSqlStr, pStmtValList);
+        this.ddlStmtBatchNum = ddlStmtBatchNum;
+    }
+
+    @Override
+    public Object apply(long value) {
+        int trackingCnt = threadBatchTrackingCntTL.get();
+        trackingCnt = trackingCnt + 1;
+        threadBatchTrackingCntTL.set(trackingCnt);
+
+        PreparedStatement stmt = (PreparedStatement) super.createDMLStatement();
+        stmt = super.setPrepStmtValues(stmt, this.pStmtValList);
+
+        try {
+            // No batch
```

Comment:
Let's discuss the batch design. There are some design approaches which will not be so adapter-specific, but will adapt to a driver and bind to the generation layer more cleanly. For later!

Reply:
I'm going to add this in a TBD list for future work